src/context/filter.tsx

```diff
@@ -0,0 +1,46 @@
+import { createContext, useCallback, useContext, useState } from "react";
+
+export type FilterContextType = {
+  field: string;
+  value: string | number;
+  setTag: (field: string, value: string | number) => void;
+  selected: any;
+};
+const FilterContext = createContext<FilterContextType | undefined>(undefined);
+
+export function FilterProvider({ children }: { children: React.ReactNode }) {
+  const [field, setField] = useState<string>("");
+  const [value, setValue] = useState<string | number>("");
+  const [selected, setSelected] = useState<any>([]);
+
+  const setTag = useCallback(
+    (field: string, value: string | number) => {
+      setField(field);
+      setValue(value);
```

Comment:
`setField` and `setValue` are no longer needed or required

Reply:
removed